DAO/MessageDAO.py

```diff
@@ -125,8 +125,9 @@ def get_num_like_photos_daily(self, pid, date, like):
     def get_trending_hashtags_day(self, date):
         cursor = self.get_cursor()
         end_date = date + relativedelta(days=1)
-        query = "select count(*) as num, hid, hashtag from hashtags_messages natural inner join messages " \
+        query = "with trending as (select count(*) as num, hid from hashtags_messages natural inner join messages " \
                 "natural inner join hashtags where messages.created_on > %s and messages.created_on < %s " \
-                "group by hid order by num desc"
+                "group by hid order by num desc limit 10)" \
+                "select hashtag from hashtags inner join trending on trending.hid = hashtags.hid"
```

Comment:
Does using a NATURAL INNER JOIN here not give you the same results?

Reply:
surprisingly no... It wasn't working when doing it that way. I'm thinking it doesn't work with subqueries but it doesn't make sense

Reply:
Could you run both queries separately and post pictures of tables? What if you switch the position of num and hid?

Reply:
... Se me habia quedado el natural INNER join. In my defense I was using sql server yesterday

Reply:
It happens